### PR TITLE
(bug) only allows users to edit hunts they created, closes #119

### LINKED ISSUE #119: Prevent creators from editing each others hunts
Currently the logic for hunt edit does not validate who is editing the hunt allowing a creators to potentially override another players hunt.

### PR DESCRIPTION
**Pull request by wettowelreactor**

### CODE REVIEW

**Comment by jcrben** [comment]
Ideally, users would get feedback before they submit a post request (maybe through an authentication check when they try to open other users' hunts) - I think this is OK in the meantime but the current situation still seems problematic and worth leaving in backlog.

**Comment by wettowelreactor** [comment]
The only way a user would get the wrong hunt is my manually entering an ID for another users hunt. There is no path within the current UI that would allow a user to edit another users hunt directly. This protects against direct manipulation. 

There is still the possibility for a user to open another users hunt in edit view but that is a separate concern. As all hunts are currently public anyways this other issue seems minor at the moment.

File: Server/src/createMiddleware.js [hunk 1]
```diff
@@ -166,7 +166,11 @@ module.exports.addSubdomain = function(app) {
     var huntid = req.params.huntid;
     var hunt = req.body;
     hunt._id = huntid;
-    hunt.creatorId = req.user;
+    if (hunt.creatorId !== req.user) {
+      console.log('user tried to modify a hunt the don\'t own');
```
Comment (MichaelJGW) [inline Server/src/createMiddleware.js]: Do you mean "user tried to modify a hunt they don\'t own"

**Comment by jcrben** [comment]
OK, that makes sense.

**Pull request merged**